test/storage/storage_test.cpp

```diff
@@ -0,0 +1,266 @@
+#include <gtest/gtest.h>
+
+#include "veriblock/arith_uint256.hpp"
+#include "veriblock/storage/block_repository_inmem.hpp"
+#include "veriblock/storage/block_repository_rocks_setup.hpp"
+#include "veriblock/uint.hpp"
+
+using namespace VeriBlock;
+
+enum class TEST_RUNS { INMEM = 0, ROCKS_BTC, ROCKS_VBK };
+
+struct BlockBasic {
+  using hash_t = uint256;
+  using height_t = int;
+
+  hash_t getHash() const { return hash; }
+
+  hash_t hash{};
+  int content = 0;
+
+  std::string toRaw() const {
+    WriteStream stream;
+    stream.writeLE<height_t>(content);
+    stream.write(hash);
+    return std::string(reinterpret_cast<const char*>(stream.data().data()),
+                       stream.data().size());
+  }
+
+  static BlockBasic fromRaw(const std::string& bytes) {
+    ReadStream stream(bytes);
+    BlockBasic block;
+    block.content = stream.readLE<height_t>();
+    block.hash = stream.read(sizeof(hash_t));
+    return block;
+  }
+
+  bool operator==(const BlockBasic& b) const {
+    return hash == b.hash && content == b.content;
+  }
+};
+
+// DB name
+static const std::string dbName = "db-test";
+
+class StorageTest : public testing::TestWithParam<TEST_RUNS> {
+ public:
+  std::shared_ptr<BlockRepository<BlockBasic>> repo;
+
+  StorageTest() : database(dbName) {
+    TEST_RUNS repoSelector = GetParam();
+    if (repoSelector == TEST_RUNS::INMEM) {
+      inmemRepo = std::make_shared<BlockRepositoryInmem<BlockBasic>>();
+    } else {
+      rocksdb::Status s = database.Open();
+      rocksBtcRepo = database.repoBtc;
+      rocksVbkRepo = database.repoVbk;
+    }
+
+    switch (repoSelector) {
+      case TEST_RUNS::INMEM:
+        repo = inmemRepo;
+        break;
+      case TEST_RUNS::ROCKS_BTC:
+        repo = rocksBtcRepo;
+        break;
+      case TEST_RUNS::ROCKS_VBK:
+        repo = rocksVbkRepo;
+        break;
+    }
+    repo->clear();
+  }
+
+ protected:
+  BlockRepositoryRocksInstance<BlockBasic, BlockBasic> database;
+  std::shared_ptr<BlockRepositoryInmem<BlockBasic>> inmemRepo;
+  std::shared_ptr<BlockRepositoryRocks<BlockBasic>> rocksBtcRepo;
+  std::shared_ptr<BlockRepositoryRocks<BlockBasic>> rocksVbkRepo;
```

Comment:
why do you have all 3 here?

Reply:
I tried to put them outside the Test class but for some reason RocksDB is not initialized properly this way.